Library/Homebrew/cask/dsl.rb

```diff
@@ -181,7 +181,7 @@ def url(*args, **options)
 
       set_unique_stanza(:url, args.empty? && options.empty? && !block_given?) do
         if block_given?
-          LazyObject.new { URL.new(*yield, from_block: true, caller_location: caller_location) }
+          LazyObject.new { URL.new(*yield, *args, **options, from_block: true, caller_location: caller_location) }
```

Comment:
Does this work?

```suggestion
          url, **options = yield
          LazyObject.new { URL.new(url, **options, from_block: true, caller_location: caller_location) }
```

Reply:
I don't think so. I'm not that familiar with `yield` so I'm not sure why this didn't work in the first place. 

```
Error: /usr/local/Homebrew/Library/Homebrew/cask/dsl.rb:184: syntax error, unexpected **arg, expecting '='
          url, **options = yield
               ^~
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
```

Reply:
update: this does work!

```ruby
          yielded = *yield
          url = yielded[0]
          opts = yielded[1] || {}
          opts[:from_block] = true
          opts[:caller_location] = caller_location
          LazyObject.new { URL.new(url, opts) }
```

Reply:
@reitermarkus would you prefer this s tyle?

Reply:
if I do it this way, this pull request can be closed without merging: https://github.com/Homebrew/homebrew-cask/pull/99447

Reply:
This should work:

```ruby
*args = yield
options = args.last.is_a?(Hash) : args.pop : {}
LazyObject.new { URL.new(*args, **options, from_block: true, caller_location: caller_location) }
```

Reply:
fixed, thanks!